npm/npm.go

```diff
@@ -177,6 +178,7 @@ func (npMgr *NetworkPolicyManager) Start(stopCh <-chan struct{}) error {
 		return fmt.Errorf("Network policy informer failed to sync")
 	}
 
+	go npMgr.reconcileChains()
```

Comment:
My main question about this is, are there any other goroutines calling iptables? If so, what is the consequence of having iptables commands being run simultaneously? Not necessarily asking about iptables being thread safe since each invocation might have its own process, but interleaved invocations (goroutine A and B both call `CheckAndAddForwardChain` at the same time, which in turn calls iptables multiple times). Do those blocks need to perform atomically?

Reply:
Iptables have inbuilt locking mechanism and NPM has a 60 sec wait to acquire the lock and then perform iptables action, so we do not have to worry about concurrency in issuing iptable commands